templates/openharmony/entry/src/main/ets/cocos/oh-adapter/sys-ability-polyfill.js

```diff
@@ -89,75 +109,113 @@ try {
+        {interval: 200000000}
     );
 } catch (err) {
     sDeviceMotionValues[6] = 0;
     sDeviceMotionValues[7] = 0;
     sDeviceMotionValues[8] = 0;
 }
-// Keep this, in the master branch, this interface has been replaced.
-//try {
-//    sensor.on(sensor.SensorId.ACCELEROMETER, function (data) {
-//        sDeviceMotionValues[0] = data.x;
-//        sDeviceMotionValues[1] = data.y;
-//        sDeviceMotionValues[2] = data.z;
-//    },
-//        { interval: 10000000000 }
-//    );
-//} catch (err) {
-//    sDeviceMotionValues[0] = 0;
-//    sDeviceMotionValues[1] = 0;
-//    sDeviceMotionValues[2] = 0;
-//}
-//
-//try {
-//    sensor.on(sensor.SensorId.LINEAR_ACCELEROMETER,function(data){
-//        sDeviceMotionValues[3] = data.x;
-//        sDeviceMotionValues[4] = data.y;
-//        sDeviceMotionValues[5] = data.z;
-//    },
-//        {interval: 10000000000}
-//    );
-//} catch (err) {
-//    sDeviceMotionValues[3] = 0;
-//    sDeviceMotionValues[4] = 0;
-//    sDeviceMotionValues[5] = 0;
-//}
-//try {
-//    sensor.on(sensor.SensorId.GYROSCOPE,function(data){
-//        sDeviceMotionValues[6] = data.x;
-//        sDeviceMotionValues[7] = data.y;
-//        sDeviceMotionValues[8] = data.z;
-//    },
-//        {interval: 10000000000}
-//    );
-//} catch (err) {
-//    sDeviceMotionValues[7] = 0;
-//    sDeviceMotionValues[8] = 0;
-//    sDeviceMotionValues[9] = 0;
-//}
 
 globalThis.getDeviceMotionValue = function () {
     return sDeviceMotionValues;
 }
+
+
+globalThis.getNetworkType = function () {
+    let netHandle = connection.getDefaultNetSync();
+    if(netHandle && netHandle.netId != 0) {
+        let result = connection.getNetCapabilitiesSync(netHandle);
+        if (result && result.bearerTypes) {
+            return result.bearerTypes[0];
+        }
+    }
+    return -1;
+}
+
+globalThis.vibrate = function (duration) {
+    console.log('begin to vibrate, duration is.' + duration);
+    try {
+        vibrator.startVibration({
+            type: 'time',
+            duration: duration * 1000
+        }, {
+            id: 0,
+            usage: 'alarm'
+        }, (error) => {
+            if (error) {
+                console.error('vibrate fail, error.code: ' + error.code + 'error.message: ', + error.message);
+                return error.code;
+            }
+            console.log('Vibration start sucessful.');
+            return 0;
+        });
+      } catch (err) {
+        console.error('errCode: ' + err.code + ' ,msg: ' + err.message);
+      }
+}
+
+globalThis.terminateProcess = function () {
+    pro.exit(0);
+}
+
+globalThis.initScreenInfo = function () {
+    display.getDefaultDisplaySync().getCutoutInfo().then((data) => {
+        if (data.boundingRects.length == 0) {
+            return;
+        }
+
+        cutout.left = data.boundingRects[0].left;
+        cutout.top = data.boundingRects[0].top;
+        cutout.width = data.boundingRects[0].width;
+        cutout.height = data.boundingRects[0].height;
```

Comment:
Define a temp variable to avoid invoking `data.boundingRects[0]` many times.

Reply:
OK